cpp/src/arrow/sparse_tensor.cc

```diff
@@ -128,17 +160,7 @@ class SparseTensorConverter<TYPE, SparseCOOIndex>
             *indices++ = static_cast<c_index_value_type>(coord[i]);
           }
         }
-
-        // increment index
-        ++coord[ndim - 1];
-        if (n > 1 && coord[ndim - 1] == shape[ndim - 1]) {
-          int64_t d = ndim - 1;
-          while (d > 0 && coord[d] == shape[d]) {
-            coord[d] = 0;
-            ++coord[d - 1];
-            --d;
-          }
-        }
+        IncrementIndex(coord, n, shape, ndim);
```

Comment:
I guess that `n` can be removed from the parameters of `IncrementIndex` by rewriting as follow.

```
if (n > 1) {
  IncrementIndex(coord, shape, ndim);
}
```